src/lib.rs

```diff
@@ -211,3 +218,189 @@ impl fmt::Display for Error {
 }
 
 impl std::error::Error for Error {}
+
+/// Run the provided actor.
+///
+/// This is the primary event loop of an actor which takes messages out of the mailbox and hands
+/// them to the actor.
+pub async fn run<A>(mut mailbox: Mailbox<A>, mut actor: A) -> A::Stop
+where
+    A: Actor,
+{
+    actor.started(&mut mailbox).await;
+
+    while let ControlFlow::Continue(()) = yield_once(&mut mailbox, &mut actor).await {}
+
+    return actor.stopped().await;
+}
```

Comment:
I mean, this is pretty cool.

Reply:
I feel like this should also be possible on master, actually 🤔

Reply:
What type are you going to pass into `Actor::started`? You'd have to do something like #76 I think.

Reply:
I mean that the `loop` can be replaced by the `while let` - I think the if is still necessary unfortunately

Reply:
> I mean that the `loop` can be replaced by the `while let` - I think the if is still necessary unfortunately

Ah yes, the `while let` is also possible on master :)